From 0f1397b39ce3d69eae059075606f704c6803d04c Mon Sep 17 00:00:00 2001
From: Yunqiang Hu <yunhu@qti.qualcomm.com>
Date: Thu, 14 May 2026 15:46:17 +0800
Subject: [PATCH] routing: fix subscriber query double call UB

Description: Cache subscriber list locally to avoid UB from iterating two different temporaries.
Signed-off-by: Yunqiang Hu <yunhu@qti.qualcomm.com>
---
 implementation/routing/src/routing_manager_base.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/implementation/routing/src/routing_manager_base.cpp b/implementation/routing/src/routing_manager_base.cpp
index 51370e7d..3aa7b2ed 100644
--- a/implementation/routing/src/routing_manager_base.cpp
+++ b/implementation/routing/src/routing_manager_base.cpp
@@ -53,7 +53,9 @@ routing_manager_base::routing_manager_base(routing_manager_host *_host) :
 void routing_manager_base::debounce_timeout_update_cbk(const boost::system::error_code &_error, const std::shared_ptr<vsomeip_v3::event> &_event, client_t _client, const std::shared_ptr<debounce_filter_impl_t> &_filter) {
     if (!_error) {
         std::lock_guard<std::mutex> its_lock(debounce_mutex_);
-        if (_event && (std::find(_event->get_subscribers().begin(), _event->get_subscribers().end(), _client) != _event->get_subscribers().end()) && _filter) {
+        const auto its_subs = _event->get_subscribers();
+        if (_event && (std::find(its_subs.begin(), its_subs.end(), _client)
+                != its_subs.end()) && _filter) {
             bool is_elapsed{false};
             auto its_current = std::chrono::steady_clock::now();
 
-- 
2.34.1

